modules/storages/app/views/storages/project_settings/_form.html.erb

```diff
@@ -0,0 +1,79 @@
+    <%= f.select :storage_id,
+        available_storages.map { |storage| ["#{storage.name} (#{storage.short_provider_type})", storage.id] },
+                  selected: available_storages.first.id.to_s,
+                  class: '-bold',
+                  container_class: '-slim' %>
+    <% if User.current.admin? %>
+      <span class="form--field-instructions">
+        <%= t(:"storages.instructions.setting_up_additional_storages")  %>
+        <%= link_to "#{t(:label_administration)} / #{t(:project_module_storages)}", controller: '/storages/admin/storages', action: :index %>.
+      </span>
+    <% end %>
+  <% when 'edit' %>
+    <%= f.label :storage_id, class: "form--label -bold" do %>
+      <%= t(:"storages.label_storage") %>
+    <% end %>
+    <div class="form--field-container -horizontal">
+      <span><%= @project_storage.storage.name %></span>
+    </div>
+  <% end %>
+</div>
+
+<% if OpenProject::FeatureDecisions.storage_project_folders_active? %>
+  <div class="form--field -required">
+    <label class="form--label -bold">Project folder</label>
+    <div class="form--field-container -horizontal">
+      <%= f.label :project_folder_mode, class: "form--label-with-radio-button" do %>
+        <%= f.radio_button :project_folder_mode, 'inactive', no_label: true %>
+        <%= t(:"storages.label_no_specific_folder") %>
+      <% end %>
+    </div>
+    <p class="form--field-instructions">
+      <%= t(:"storages.instructions.no_specific_folder")  %>
+    </p>
+
+    <div class="form--field-container -horizontal">
+      <%= f.label :project_folder_mode, class: "form--label-with-radio-button" do %>
+        <%= f.radio_button :project_folder_mode, 'manual', no_label: true %>
+        <%= t(:"storages.label_existing_manual_folder") %>
+      <% end %>
+    </div>
+    <p class="form--field-instructions -xwide">
+      <%= t(:"storages.instructions.existing_manual_folder")  %>
+    </p>
+  </div>
+
+  <div style="display:none;" id="storages_manual_folder_selection" class="form--field -required">
+    <label class="form--label -bold"> <%= t(:"storages.label_existing_manual_folder_selection") %> </label>
+    <div class="form--field-container -horizontal">
+      <%= f.text_field :project_folder_id,
+                      placeholder: t(:"storages.placeholder_existing_folder_selection"),
+                      container_class: "-middle",
+                      no_label: true
+      %>
+    </div>
+  </div>
+
+  <%= nonced_javascript_tag do -%>
+    jQuery(function($) {
+      <%# Show manual folder selection on reload when checked %>
+      $("#storages_project_storage_project_folder_mode_manual").is(':checked') && $("#storages_manual_folder_selection").show();
+
+      $("input[type='radio']").change(function() {
+        if ($(this).val() == 'manual') {
+          $("#storages_manual_folder_selection").show();
+        } else {
+          $("#storages_manual_folder_selection").hide();
+        }
+      })
+    });
+  <% end -%>
```

Comment:
Did you talk to Oliver about the dynamic scripts? There are IMHO two things, I would suggest as an improvement:

- use the meta tag for required scripts

```erb
<% content_for :header_tags do %>
  <meta name="required_script" content="project_folder_mode_selection" />
<% end %>
```

- get rid of jQuery and write a basic typescript file

If there is a good reason not to do this right now, we should do this in the next step, when this script will get much more complex.

Reply:
Good shout! As discussed, we'll pair to address the above. 👍🏾